src/main/java/com/amalitech/gpuconfigurator/controller/AuthController.java

```diff
@@ -0,0 +1,89 @@
+package com.amalitech.gpuconfigurator.controller;
+
+import com.amalitech.gpuconfigurator.dto.*;
+import com.amalitech.gpuconfigurator.model.OtpType;
+import com.amalitech.gpuconfigurator.repository.UserRepository;
+import com.amalitech.gpuconfigurator.service.MailService;
+import com.amalitech.gpuconfigurator.service.OtpService;
+import com.amalitech.gpuconfigurator.service.UserService;
+import jakarta.mail.MessagingException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/v1/auth")
+@RequiredArgsConstructor
+public class AuthController {
+    private final UserRepository userRepository;
+    private final UserService userService;
+    private final OtpService otpService;
+    private final MailService mailService;
+
+    @PostMapping("/reset-password")
+    public ResponseEntity<ResetPasswordResponseDTO> resetPassword(
+            @Validated @RequestBody ResetPasswordDTO resetPasswordDto
+    ) {
+        if (userRepository.existsByEmail(resetPasswordDto.getEmail())) {
```

Comment:
It would better if this was done in a service rather than a controller

Reply:
i think it would be also nice if we used the Optional<User> and throw custom error instead of try catch

Reply:
it would be nice if we create custom errors that return Response entities and throw them rather